### ISSUE TITLE
Crop2D just works with ortho Z (2)

### ISSUE THREAD

**Issue by institutohidraulica** [tmontaigu/CloudCompare-PythonPlugin#77]
I am trying to perform a crop2d based on the example file, but when shifting from Z plane (ortho int = 2) to Y plane (int =1), with a new polyline with vertices (X1, 0, Z1), (X2, 0, Z2), (X3, 0, Z3), (X4, 0, Z4), it always return that no points are within the cropped section, and this is not true,
Can it be a bug of the crop2D implementation?
I also noticed that in previous versions of CC it was also incorrect: https://www.cloudcompare.org/forum/viewtopic.php?t=1805

**Comment by tmontaigu** [comment]
Given that the python crop2d simply calls the cloudcompare's function then if there's a problem its located in cloudcompare.

But first, I think you polyline should be 2D, so (X1, Z1, 0)

From a quick look at the code, with orto dim = 2, the point cloud X coordinates becomes the Z coordinates, and the Y coordinates becomes X. So maybe you have to swap your  X, Z of your polyline

**Comment by institutohidraulica** [comment]
Yes, I also shifted the X-Z coordinates, BUT I did it just on the CCVector3 offset. 
Based on the example, I took --> bbMin + cccorelib.CCVector3(0, diag.y / 2, 0)
being cccorelib.CCVector3(x1, y1, 0)
and in order to so into Y ortho plane I shifted to cccorelib.CCVector3(z1, x1, 0)
BUT I forget to shift also bbMin to (Z, X, Y)
This is why I can not get any point. I now shift also bbMin to Z,X,Y and now is working correctly!
Thanks for your help!
